src/detect-engine.c

```diff
@@ -2606,6 +2606,9 @@ DetectEngineCtx *DetectEngineCtxInitWithPrefix(const char *prefix, uint32_t tena
 static void DetectEngineCtxFreeThreadKeywordData(DetectEngineCtx *de_ctx)
 {
     HashListTableFree(de_ctx->keyword_hash);
+#if UNITTESTS
+    de_ctx->keyword_hash = NULL;
```

Comment:
Double free Happens only in unit tests, right ? Just double checking

Reply:
And why was CI green if there was this double free ?

Reply:
Yes, this is a unit test only situation.

The double free would occur in DetectUnregsiterThreadCtxFuncs iff keyword hash wasn't nullified.

Reply:
And why was CI green if there was this double free ?